ui-v2/app/controllers/dc/intentions/edit.js

```diff
@@ -9,24 +9,36 @@ export default Controller.extend({
     this.form = this.builder.form('intention');
   },
   setProperties: function(model) {
-    const sourceName = get(model.item, 'SourceName');
-    const destinationName = get(model.item, 'DestinationName');
-    let source = model.items.findBy('Name', sourceName);
-    let destination = model.items.findBy('Name', destinationName);
+    let source = model.services.findBy('Name', model.item.SourceName);
+
     if (!source) {
-      source = { Name: sourceName };
-      model.items = [source].concat(model.items);
+      source = { Name: model.item.SourceName };
+      model.services = [source].concat(model.services);
     }
+    let destination = model.services.findBy('Name', model.item.DestinationName);
     if (!destination) {
-      destination = { Name: destinationName };
-      model.items = [destination].concat(model.items);
+      destination = { Name: model.item.DestinationName };
+      model.services = [destination].concat(model.services);
+    }
+
+    let sourceNS = model.nspaces.findBy('Name', model.item.SourceNS);
+    if (!sourceNS) {
+      sourceNS = { Name: model.item.SourceNS };
+      model.nspaces = [sourceNS].concat(model.nspaces);
+    }
+    let destinationNS = model.nspaces.findBy('Name', model.item.DestinationNS);
+    if (!destinationNS) {
+      destinationNS = { Name: model.item.DestinationNS };
+      model.nspaces = [destinationNS].concat(model.nspaces);
     }
     this._super({
       ...model,
       ...{
         item: this.form.setData(model.item).getData(),
         SourceName: source,
         DestinationName: destination,
+        SourceNS: sourceNS,
+        DestinationNS: destinationNS,
```

Comment:
do these just get ignored by the OSS binary?

Reply:
I think so yeah, we've tested it against the binary and everything works a-ok. @mkeeler just to double check if I POST these values to OSS they are noop's right?